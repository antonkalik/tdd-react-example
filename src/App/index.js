import React from "react";
import { Header } from "src/components/Header";
import { Body } from "src/components/Body";
import { Footer } from "src/components/Footer";
import { useCards } from "src/hooks/useCards";
import { StyledApp } from "./style";

export function App() {
  const result = useCards();

  return (
    <StyledApp data-testid="app">
      <Header />
      <Body cards={result.cards} />
      <Footer />
    </StyledApp>
  );
}
