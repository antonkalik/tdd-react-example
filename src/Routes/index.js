import { Routes as ReactRoutes, Route, Outlet } from "react-router-dom";
import { HomeView } from "src/views/HomeView";
import { CardsView } from "src/views/CardsView";
import { CardView } from "src/views/CardView";
import { AboutView } from "src/views/AboutView";
import { MainLayout } from "src/components/MainLayout";

export const Routes = () => {
  return (
    <ReactRoutes>
      <Route path="/" element={<MainLayout />}>
        <Route index element={<HomeView />} />
        <Route path="/cards" element={<Outlet />}>
          <Route index element={<CardsView />} />
          <Route path=":id" element={<CardView />} />
        </Route>
        <Route path="/about" element={<AboutView />} />
      </Route>
      <Route path="*" element={<h1>404 - Not Found</h1>} />
    </ReactRoutes>
  );
};
