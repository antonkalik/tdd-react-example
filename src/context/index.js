import React from "react";

export const initialState = { cards: [] };

export function reducer(state, action) {
  switch (action.type) {
    case "ADD_CARD":
      return {
        cards: action.payload.title
          ? [
              ...state.cards,
              {
                ...action.payload,
                id: state.cards.length + 1,
              },
            ]
          : state.cards,
      };
    case "REMOVE_CARD":
      return {
        cards: state.cards.filter((card) => card.id !== action.payload.id),
      };
    case "RESET":
      return initialState;
    default:
      return state || initialState;
  }
}

export const AppContext = React.createContext();

const StoreProvider = ({ children }) => {
  const value = React.useReducer(reducer, initialState);

  return <AppContext.Provider value={value}>{children}</AppContext.Provider>;
};

export default StoreProvider;
