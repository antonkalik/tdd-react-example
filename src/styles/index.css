html,
body {
  background-color: #222;
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

.app {
  padding: 20px;
}

a,
p,
h1,
h2,
h3,
h4,
h5,
h6,
div {
  color: #fff;
}
