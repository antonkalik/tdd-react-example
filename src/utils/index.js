export const initCards = Array.from({ length: 5 }, (_, i) => ({
  title: `Test Card ${i + 1}`,
  id: i + 1,
}));
