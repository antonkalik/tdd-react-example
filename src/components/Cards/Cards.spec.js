import { render, screen } from "@testing-library/react";
import { initCards } from "src/utils";
import { Cards } from "./index";

describe("<Cards />", () => {
  it("should render cards", () => {
    render(<Cards />);
    expect(screen.getByTestId("cards")).toMatchSnapshot();
  });

  it("should render main title", () => {
    render(<Cards />);
    expect(screen.getByText("All Cards")).toBeInTheDocument();
  });

  it("should render message with no cards", () => {
    render(<Cards />);
    expect(screen.getByText("No cards")).toBeInTheDocument();
  });

  it("should render one card", () => {
    render(
      <Cards
        cards={[
          {
            id: 1,
            title: "Card 1",
          },
        ]}
      />
    );
    const element = screen.getByTestId("card");
    expect(element).toBeInTheDocument();
    expect(screen.getByTestId("cards")).toMatchSnapshot();
    expect(element).toHaveTextContent("Card 1");
  });

  it("should render multiple cards", () => {
    render(<Cards cards={initCards} />);
    const elements = screen.getAllByTestId("card");
    expect(elements).toHaveLength(initCards.length);
  });
});
