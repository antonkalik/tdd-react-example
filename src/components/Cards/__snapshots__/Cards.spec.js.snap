// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards /> should render cards 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c0 > h3 {
  width: 100%;
  text-align: left;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c1 > h4 {
  width: 100%;
  text-align: center;
}

<div
  class="c0"
  data-testid="cards"
>
  <h3>
    All Cards
  </h3>
  <div
    class="c1"
    data-testid="cards-list"
  >
    <h4>
      No cards
    </h4>
  </div>
</div>
`;

exports[`<Cards /> should render one card 1`] = `
.c2 {
  border: 1px solid #ccc;
  border-radius: 4px;
  max-width: 200px;
  padding: 8px;
  margin: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 h5 {
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c0 > h3 {
  width: 100%;
  text-align: left;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c1 > h4 {
  width: 100%;
  text-align: center;
}

<div
  class="c0"
  data-testid="cards"
>
  <h3>
    All Cards
  </h3>
  <div
    class="c1"
    data-testid="cards-list"
  >
    <div
      class="c2"
      data-testid="card"
    >
      <h5>
        Card 1
      </h5>
    </div>
  </div>
</div>
`;
